### PR TITLE
#69914 abstract reader creation

### PR DESCRIPTION
**Pull request by tabman83**

### CODE REVIEW

File: src/Tests/CaptainHook.Tests/Configuration/KeyVaultProviderTests.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace CaptainHook.Tests.Configuration
 {
     public class KeyVaultProviderTests
     {
-        [Fact]
+        [Fact(Skip = "Superseded by ConfigLoaderTests.ConfigNotEmpty")]
```
Comment (chomick) [inline src/Tests/CaptainHook.Tests/Configuration/KeyVaultProviderTests.cs:15]: Do we need that test/code for anything if duplicated?

Comment (tabman83) [reply src/Tests/CaptainHook.Tests/Configuration/KeyVaultProviderTests.cs:15]: Nope. I just thought it was good practice to deprecate by skipping it before removing it entirely. Can be removed now or in the next PR.

File: src/CaptainHook.DirectorService/DirectorService.cs [hunk 2]
```diff
@@ -58,132 +51,28 @@ protected override async Task RunAsync(CancellationToken cancellationToken)
...
-                        cancellationToken);
-                }
-
-                foreach (var (key, subscriber) in _subscriberConfigurations)
-                {
-                    if (cancellationToken.IsCancellationRequested) return;
-
-                    await RecreateServiceForSubscriberAsync(subscriber, serviceList, cancellationToken);
+                    var description = new ServiceCreationDescription(ServiceNaming.EventHandlerServiceFullName, ServiceNaming.EventHandlerActorServiceType);
+                    await _fabricClientWrapper.CreateServiceAsync(description, cancellationToken);
```
Comment (chomick) [inline src/CaptainHook.DirectorService/DirectorService.cs:60]: Should that not happen for each subscriber config?

File: src/CaptainHook.DirectorService/FabricClientWrapper.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using System.Threading.Tasks;
+
+namespace CaptainHook.DirectorService
+{
+    /// <summary>
+    /// A simple wrapper around FabricClient
+    /// </summary>
+    public class FabricClientWrapper : IFabricClientWrapper
+    {
+        private readonly FabricClient fabricClient;
```
Comment (chomick) [inline src/CaptainHook.DirectorService/FabricClientWrapper.cs:18]: Can you use _ prefix for local fields?

File: src/CaptainHook.DirectorService/SubscriptionManager.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CaptainHook.DirectorService
+{
+    public class SubscriptionManager
+    {
+        private readonly IFabricClientWrapper fabricClientWrapper;
```
Comment (chomick) [inline src/CaptainHook.DirectorService/SubscriptionManager.cs:13]: Please use "_" prefix

File: src/CaptainHook.DirectorService/FabricClientWrapper.cs (hunk 3 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/FabricClientWrapper.cs:18]: Not a problem. Would be nice to define one convention in a project anyway.

**Pull request merged**